gdal/ogr/ogrsf_frmts/amigocloud/ogramigoclouddatasource.cpp

```diff
@@ -356,12 +357,30 @@ OGRLayer   *OGRAmigoCloudDataSource::ICreateLayer( const char *pszNameIn,
         return NULL;
     }
 
-/* -------------------------------------------------------------------- */
-/*      Do we already have this layer?  If so, should we blow it        */
-/*      away?                                                           */
-/* -------------------------------------------------------------------- */
     CPLString osName(pszNameIn);
 
+    if (nLayers > 1) {
+        // We cannot have more then one destination dataset (layer).
```

Comment:
TestCapability(ODsCCreateLayer) should also return FALSE in that case

Reply:
If I do that, It doesn't work:
 ./apps/ogr2ogr  -f AmigoCloud "AmigoCloud:15276 datasets=100342" ~/Downloads/Visits.geojson
ERROR 1: Layer 'Visits' does not already exist in the output dataset, and cannot be created by the output driver.

Reply:
Well as you return NULL in ICreateLayer(), that can't work either... Or perhaps you prefer the explicit "Multiple destination datasets are not allowed" error message. OK. The way the Amigocloud driver manages layers is a bit at odds with other drivers, but I won't get into your way ;-)

Reply:
I think this whole confusion is started with me trying to use existing amigocloud destination layer. Perhaps I am missing a big picture? My thinking is iCreateLayer() returns a new layer, or existing one, based on different conditions. Is there another way handling existing destination layers?

Reply:
Please look at other driver implementations

Reply:
Is there a documentation on architecture? How for instance gdal opens existing destination layer without ICreateLayer() being involved?